### PR TITLE
Fix PSVersion parameter version checks and error messages for PowerShell core remoting

### PR DESCRIPTION
**Pull request by PaulHigin**
# PR Summary

This PR addresses issue #16062.  

## PR Context

Windows PowerShell supports remoting endpoint configurations that run on versions 2,3,4,5 of PowerShell.  However, the PowerShell core WinRM plugin only supports endpoints that run in the currently installed version of PowerShell core, and ignores the `-PSVersion` parameter.  This change removes the old supported version checks for the `-PSVersion` parameter on the configuration cmdlets and now throws a `not supported` error.

Also, `Start-Job` used to support running jobs in multiple WindowsPowerShell versions via the `-PSVersion` parameter, but now only supports version `5.1` in PowerShell core.  So this fix updates the version check and error message to provide correct information.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by daxian-dbw** [comment]
@TravisEz13 and @SteveL-MSFT please review when you have time, thanks!

File: src/System.Management.Automation/engine/remoting/commands/StartJob.cs [hunk 1]
```diff
@@ -512,10 +512,12 @@ public virtual Version PSVersion
 
             set
             {
-                RemotingCommandUtil.CheckPSVersion(value);
-
-                // Check if specified version of PowerShell is installed
-                RemotingCommandUtil.CheckIfPowerShellVersionIsInstalled(value);
+                // PSVersion value can only be 5.1 for Start-Job.
+                if (value.Major != 5 || value.Minor != 1)
```
Comment (JamesWTruher) [inline src/System.Management.Automation/engine/remoting/commands/StartJob.cs:516]: from a readability perspective, would it be more direct to the following:
```suggestion
                if (!(value.Major == 5 && value.Minor == 1))
```

File: src/System.Management.Automation/engine/remoting/commands/StartJob.cs [hunk 2]
```diff
@@ -512,10 +512,12 @@ public virtual Version PSVersion
 
             set
             {
-                RemotingCommandUtil.CheckPSVersion(value);
-
-                // Check if specified version of PowerShell is installed
-                RemotingCommandUtil.CheckIfPowerShellVersionIsInstalled(value);
+                // PSVersion value can only be 5.1 for Start-Job.
+                if (!(value.Major == 5 && value.Minor == 1))
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/remoting/commands/StartJob.cs:516]: Do we need to check if this is Windows and return a different error if used on Linux/macOS?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1 [hunk 3]
```diff
@@ -243,3 +243,16 @@ Describe "Ampersand background test" -Tag "CI","Slow" {
         }
     }
 }
+
+Describe "Start-Job with -PSVersion parameter" -Tag "CI" {
+
+    It "Verifies that -PSVersion is not supported except for version 5.1" -Skip:(-not $IsWindows) {
```
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1:249]: If we aren't doing a different error for non-Windows, then this test should also pass on non-Windows, right?

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**